From 6e225839ac6a2c0d3c62a991f5d1b006a4af3742 Mon Sep 17 00:00:00 2001
From: "Liu,Dalei" <daleix.liu@intel.com>
Date: Tue, 9 Jun 2015 15:01:10 +0800
Subject: [PATCH 417/462] pm: Skip atomisp fwload in COS

This patch exposes a cmdline parameter which is set in COS
to skip the driver wait for loading atomisp fw.This is done
to prevent the driver wait for isp firmware load , thus
preventing COS boot

Change-Id: I0b33d86af06cfe6e03fba4d62c77bb0b69099340
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-38642
Signed-off-by: Liu,Dalei <daleix.liu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/59481
Approver: Yunliang Ding <yunliang.ding@intel.com>
Reviewed-by: Yunliang Ding <yunliang.ding@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: IMIN (SC)
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c       | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
index 8f4483a..93af334 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
@@ -65,6 +65,10 @@
 #define ATOMISP_INTERNAL_PM	(IS_MOFD || IS_BYT || IS_CHT)
 #endif
 
+static uint skip_fwload = 0;
+module_param(skip_fwload, uint, 0644);
+MODULE_PARM_DESC(skip_fwload, "Skip atomisp firmware load for COS");
+
 /* set reserved memory pool size in page */
 unsigned int repool_pgnr;
 module_param(repool_pgnr, uint, 0644);
@@ -1153,6 +1157,9 @@ atomisp_load_firmware(struct atomisp_device *isp)
 	int rc;
 	char *fw_path = NULL;
 
+	if (skip_fwload)
+		return NULL;
+
 #if defined(ATOMISP_FWNAME)
 	fw_path = ATOMISP_FWNAME;
 #else
-- 
1.9.1

